app/src/main/java/com/zeapo/pwdstore/autofill/oreo/ui/AutofillFilterActivity.kt

```diff
@@ -0,0 +1,199 @@
+
+        fun makeMatchAndDecryptFileIntentSender(
+            context: Context,
+            formOrigin: FormOrigin
+        ): IntentSender {
+            val intent = Intent(context, AutofillFilterView::class.java).apply {
+                when (formOrigin) {
+                    is FormOrigin.Web -> putExtra(EXTRA_FORM_ORIGIN_WEB, formOrigin.identifier)
+                    is FormOrigin.App -> putExtra(EXTRA_FORM_ORIGIN_APP, formOrigin.identifier)
+                }
+            }
+            return PendingIntent.getActivity(
+                context,
+                matchAndDecryptFileRequestCode++,
+                intent,
+                PendingIntent.FLAG_CANCEL_CURRENT
+            ).intentSender
+        }
+    }
+
+    private val dataSource = dataSourceOf()
+    private val preferences by lazy { PreferenceManager.getDefaultSharedPreferences(this) }
+    private val sortOrder
+        get() = PasswordRepository.PasswordSortOrder.getSortOrder(preferences)
+
+    private lateinit var formOrigin: FormOrigin
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_oreo_autofill_filter)
+        setFinishOnTouchOutside(true)
+
+        val params = window.attributes
+        params.height = (HEIGHT_PERCENTAGE * resources.displayMetrics.heightPixels).toInt()
+        params.width = (WIDTH_PERCENTAGE * resources.displayMetrics.widthPixels).toInt()
+        window.attributes = params
+
+        if (intent?.hasExtra(AutofillManager.EXTRA_CLIENT_STATE) != true) {
+            e { "AutofillFilterActivity started without EXTRA_CLIENT_STATE" }
+            finish()
+            return
+        }
+        formOrigin = when {
+            intent?.hasExtra(EXTRA_FORM_ORIGIN_WEB) == true -> {
+                FormOrigin.Web(intent!!.getStringExtra(EXTRA_FORM_ORIGIN_WEB)!!)
+            }
+            intent?.hasExtra(EXTRA_FORM_ORIGIN_APP) == true -> {
+                FormOrigin.App(intent!!.getStringExtra(EXTRA_FORM_ORIGIN_APP)!!)
+            }
+            else -> {
+                e { "AutofillFilterActivity started without EXTRA_FORM_ORIGIN_WEB or EXTRA_FORM_ORIGIN_APP" }
+                finish()
+                return
+            }
+        }
+
+        supportActionBar?.hide()
+        bindUI()
+        setResult(RESULT_CANCELED)
+    }
+
+    private fun bindUI() {
+        // setup is an extension method provided by recyclical
+        rvPassword.setup {
+            withDataSource(dataSource)
+            withItem<PasswordItem, PasswordViewHolder>(R.layout.oreo_autofill_filter_row) {
+                onBind(::PasswordViewHolder) { _, item ->
+                    title.text = item.fullPathToParent
+                    // drop the .gpg extension
+                    subtitle.text = item.name.dropLast(4)
+                }
+                onClick { decryptAndFill(item) }
+            }
+        }
+        rvPassword.apply {
```

Comment:
I personally don't see the point of using `.apply` when we're only changing one property